src/main/scala/mesosphere/marathon/core/task/tracker/impl/TaskOpProcessorImpl.scala

```diff
@@ -51,8 +51,8 @@ private[tracker] object TaskOpProcessorImpl {
         case TaskStateChange.Expunge             => Action.Expunge
         case TaskStateChange.NoChange            => Action.Noop
         case TaskStateChange.Failure(cause) =>
-          log.warn(s"Task state update failed for ${task.taskId}. Reason: $cause")
-          Action.Noop
+          log.warn(s"Task state update failed for ${task.taskId}. Reason: ${cause.getMessage}")
```

Comment:
The receiver of the `Action.Failure` handles the failure. You do not need to log it here.